ostd/libs/linux-bzimage/builder/src/encoder.rs

```diff
@@ -0,0 +1,74 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! This module is used to compress kernel ELF.
+
+use std::{
+    ffi::{OsStr, OsString},
+    io::Write,
+    str::FromStr,
+};
+
+use libflate::{deflate, gzip, zlib};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Default, Clone, PartialEq, Eq, Serialize, Deserialize)]
+pub enum CompressionFormat {
+    #[default]
+    #[serde(rename = "plain")]
+    Plain,
```

Comment:
The value `CompressionFormat::Plain` may mislead the user that we do some sort of "plain compression". I think a better name could be `PayloadEncoding::Raw`.